### PR TITLE
IQR function is added

### PR DESCRIPTION
**Pull request by Mitra-babu**
### Describe your change:
Inter quartile range function is added in the repo.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by madhavgaba — CHANGES_REQUESTED** [review]
Can be simplifiied

File: maths/inter_quartile_range.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    >>> inter_quartile_range(x=np.array([25,32,49,21,37,43,27,45,31]))
+    18.0
+    """
+    length = len(x)
+    if length == 0:
+        raise ValueError
+    x.sort()
+    q1 = find_median(x[0 : length // 2])
+    if length % 2:
+        q3 = find_median(x[(length // 2) + 1 : length])
```
[the remark this answers is not in the document]
Comment (Mitra-babu) [reply maths/inter_quartile_range.py:52]: yes, We can do this. Thank you for the suggestion @madhavgaba .

**Review by madhavgaba — APPROVED** [review]

**Comment by Mitra-babu** [comment]
@cclauss please suggest any changes if any.

File: maths/inter_quartile_range.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    >>> inter_quartile_range(x=np.array([4,1,2,3,2]))
+    2.0
+
+    >>> inter_quartile_range(x=np.array([25,32,49,21,37,43,27,45,31]))
+    18.0
+    """
+    length = len(x)
+    if length == 0:
+        raise ValueError
```
Comment (tianyizheng02) [inline maths/inter_quartile_range.py:48]: Add an error message for the ValueError

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
"interquartile" is one word, not two

File: maths/inter_quartile_range.py (hunk 2 above)
Comment (cclauss) [reply maths/inter_quartile_range.py:48]: Please add an error message for the ValueError.

File: maths/interquartile_range.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    >>> interquartile_range(x=np.array([25,32,49,21,37,43,27,45,31]))
+    18.0
+    """
+    length = len(x)
+    if length == 0:
+        raise ValueError("The list is empty. Provide a non-empty list.")
+    x.sort()
+    q1 = find_median(x[0 : length // 2])
+    half_length = (length // 2) + 1 if length % 2 else length // 2
```
Comment (cclauss) [inline maths/interquartile_range.py:52]:
```suggestion
    half_length = sum(divmod(length, 2))
```

File: maths/interquartile_range.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    range for a list of numeric.
+    :param x: The list of data point
+    :return: Inter_quartile range
+
+    >>> interquartile_range(x=np.array([4,1,2,3,2]))
+    2.0
+
+
+    >>> interquartile_range(x=np.array([25,32,49,21,37,43,27,45,31]))
+    18.0
```
Comment (cclauss) [inline maths/interquartile_range.py:45]: Please rename `x` to `items` or `values` or something else.
Please add a test for the empty list.
Please add some tests that have zero, negative numbers, floating point numbers and letters.

File: maths/interquartile_range.py (hunk 3 above)
Comment (cclauss) [inline maths/interquartile_range.py:52]:
```suggestion
    div, mod = divmod(7, 2)
    q1 = find_median(x[0 : div])
    half_length = sum((div, mod))
```

Comment (tianyizheng02) [reply maths/interquartile_range.py:52]:
```py
    div, mod = divmod(7, 2)
    q1 = find_median(x[0 : div])
```
`div, mod = divmod(length, 2)`?
`q1 = find_median(x[:div])`?

**Comment by Mitra-babu** [comment]
I'll add some more test case.

**Comment by Mitra-babu** [comment]
HI @cclauss, please review the changes once.

**Review by cclauss — APPROVED** [review]
